File: AppDemo/daggersubmodule/src/main/java/com/tencent/tvs/testdagger/ActivityScope.java
package com.tencent.tvs.testdagger;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;

import javax.inject.Scope;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Created by johnny on 2017/6/8.
 */
@Documented
@Retention(RUNTIME)
@Scope
public @interface ActivityScope {}

File: AppDemo/daggerandroid/src/main/java/com/sapphire/dagger/test/di/ActivityScope.kt
package com.sapphire.dagger.test.di

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import javax.inject.Scope

@Documented
@Retention(RetentionPolicy.RUNTIME)
@Scope
annotation class ActivityScope